_posts/2017-04-30-React-Native-for-iOS-devs.markdown

```diff
@@ -0,0 +1,671 @@
+
+
+</article>
+<article class="desktop-only" style='display: flex; flex-flow: row;'>
+<div class="desktop-only" style='flex:1; display: block;'>
+
+
+```sh
+$ tree .
+
+├── __tests__
+│   ├── index.android.js
+│   └── index.ios.js
+├── android - [snipped]
+├── app.json
+├── index.android.js
+├── index.ios.js
+├── ios
+│   ├── TrendingArtists
+│   │   ├── AppDelegate.h
+│   │   ├── AppDelegate.m
+│   │   ├── Base.lproj
+│   │   │   └── LaunchScreen.xib
+│   │   ├── Images.xcassets
+│   │   │   └── AppIcon.appiconset
+│   │   │       └── Contents.json
+│   │   ├── Info.plist
+│   │   └── main.m
+│   ├── TrendingArtists-tvOS
+│   │   └── Info.plist
+│   ├── TrendingArtists-tvOSTests
+│   │   └── Info.plist
+│   ├── TrendingArtists.xcodeproj
+│   │   ├── project.pbxproj
+│   │   ├── project.xcworkspace
+│   │   │   ├── contents.xcworkspacedata
+│   └── TrendingArtistsTests
+│       ├── Info.plist
+│       └── TrendingArtistsTests.m
+├── jsconfig.json
+├── package.json
+├── node_modules [x million files snipped]
+└── yarn.lock
+```
+
+</div>
+<div class="desktop-only" style='flex:1; display: block;'>
+
+<p><strong>What are we looking at?</strong></p>
+
+<p>First up - we have some test files, these files are unique per platform - though they do have the same code right now. In React Native imports can resolve to be different per-platform, which is why you see <code>.android.js</code> or <code>.ios.js</code>.</p>
+
+<p>The next important files are: <code>index.android.js</code> and <code>index.ios.js</code>. These are the launching point for this app, so it's in there you'll find the code for what we're seeing.</p>
+
+<p>In the <code>ios</code> folder, we have the native side of React Native. Looks pretty empty from here, but when you open the workspace you'll note that it is referring to a lot of <code>xcproject</code>s which are inside your <code>node_modules</code> folder.</p>
+
+<p>The Xcode project is really barebones, it's just an <code>AppDelegate</code> that creates a <code>UIView</code> subclass. You can see that it references <code>index.ios.js</code> which is where your JavaScript side starts.</p>
+
+<p>Then you have the <code>package.json</code> which is like an Xcodeproj + Podfile in one, and the <code>node_modules</code> folder. Which houses all your JavaScript dependencies.</p>
+
+</div>
+</article>
+<article class='post'>
+
+I'd like to show you how to make a change appear instantly. In your iOS Simulator, perform a shake gesture (<code>cmd + ctrl + z</code>) and in the React Native debug menu, hit "Enable Hot Reloading". It will trigger a reload of all your JS code again.
+
+Now you can go into your text editor and change some of the words inside `index.ios.js` - those changes will be reflected almost instantly. You can do this for almost anything, almost anywhere. We've been working [on Emission][emission] for over a year, and this is still close to instant in every part of the codebase.
```

Comment:
The speed of making a change -> seeing it reflect in the app is really key to RN, you should consider adding a sentence here that contrasts that to the earlier figure of 9 seconds, and explains how significantly it's impacted your personal development productivity.